tensorflow_datasets/structured/movielens.py

```diff
@@ -0,0 +1,422 @@
+In all datasets, the movies data and ratings data are joined on "movieId".
+The 25m dataset, latest-small dataset, and 1m dataset have the same schema
+and they do not include demographic data.
+The 25m dataset is the latest stable version of the MovieLens dataset.
+It is recommended for research purposes.
+The latest-small dataset is a small dataset with
+the same schema as 25m and 20m.
+It is updated over time by GroupLens and its checksum might need updating.
+The 20m dataset is one of the most used MovieLens datasets in academic papers
+along with the 1m dataset.
+The 100k dataset and 1m dataset are the oldest MovieLens datasets.
+They both contain demographic data and have the same schema.
+"""
+
+_DATASET_OPTIONS = ['25m', 'latest-small', '100k', '1m', '20m']
+
+
+class MovieLens(tfds.core.GeneratorBasedBuilder):
+  """MovieLens rating dataset."""
+
+  # Using BuilderConfig directly since only the name field is needed for
+  # configuration.
+  BUILDER_CONFIGS = [
+      tfds.core.BuilderConfig(
+          name='25m',
+          description=textwrap.dedent("""\
+              This dataset contains 25000095 ratings across 62423 movies.
+              These data were created by 162541 users between January 09, 1995
+              and November 21, 2019.
+              Ratings are in half-star increments.
+              Each user has rated at least 20 movies.
+              This dataset is the latest stable version
+              of the MovieLens dataset.
+              This dataset does not include demographic data.
+              The movies data and ratings data are joined on "movieId".
+              Genres are converted into sequence of labels.
+              "(no genres listed)" is considered a label itself.
+              This dataset was generated on November 21, 2019."""),
+          version='0.1.0',
+      ),
+      tfds.core.BuilderConfig(
+          name='latest-small',
+          description=textwrap.dedent("""\
+              This dataset contains 100836 ratings across 9742 movies
+              These data were created by 610 users between March 29, 1996
+              and September 24, 2018.
+              Ratings are in half-star increments.
+              Each user has rated at least 20 movies.
+              This dataset is the a subset of the full latest version
+              of the MovieLens dataset.
+              This dataset is changed and updated over time and the checksum
+              might need updating.
+              This dataset does not include demographic data.
+              The movies data and ratings data are joined on "movieId".
+              Genres are converted into sequence of labels.
+              "(no genres listed)" is considered a label itself.
+              This dataset was generated on September 26, 2018."""),
+          version='0.1.0',
+      ),
+      tfds.core.BuilderConfig(
+          name='100k',
+          description=textwrap.dedent("""\
```

Comment:
We should use consistent quotes, either `'` or `"` throughout.

Reply:
Ping :)

Reply:
I chose to use `'` but for triple quotes I chose `"""`. I know docstrings must be surrounded by `"""`. Should I change the non-docstring triple quotes to `'''`?